mlflow/store/db_migrations/versions/bd07f7e963c5_create_index_on_run_uuid_for_postgres.py

```diff
@@ -0,0 +1,30 @@
+"""create index on run_uuid for postgres
+
+Revision ID: bd07f7e963c5
+Revises: c48cb773bb87
+Create Date: 2022-03-03 10:14:34.037978
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "bd07f7e963c5"
+down_revision = "c48cb773bb87"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # This part of the migration is only relevant for PostgreSQL.
+    # As a fix for https://github.com/mlflow/mlflow/issues/3785, create indexes on run_uuid columns
+    # to speed up SQL operations.
+    bind = op.get_bind()
+    if bind.engine.name == "postgresql":
+        for table in ["params", "metrics", "latest_metrics", "tags"]:
+            op.create_index(f"index_{table}_run_uuid", table, ["run_uuid"])
```

Comment:
These are unidexed foreign keys:

| table                 | columns       | size       | constraint                           | referenced_table   |
|:----------------------|:--------------|:-----------|:-------------------------------------|:-------------------|
| runs                  | experiment_id | 8192 bytes | runs_experiment_id_fkey              | experiments        |
| ✅ tags                  | run_uuid      | 8192 bytes | tags_run_uuid_fkey                   | runs               |
| ✅ metrics               | run_uuid      | 0 bytes    | metrics_run_uuid_fkey                | runs               |
| ✅ params                | run_uuid      | 0 bytes    | params_run_uuid_fkey                 | runs               |
| experiment_tags       | experiment_id | 0 bytes    | experiment_tags_experiment_id_fkey   | experiments        |
| ✅ latest_metrics        | run_uuid      | 0 bytes    | latest_metrics_run_uuid_fkey         | runs               |
| registered_model_tags | name          | 0 bytes    | registered_model_tags_name_fkey      | registered_models  |
| model_version_tags    | name,version  | 0 bytes    | model_version_tags_name_version_fkey | model_versions     |

This migration script updates rows marked with ✅.

Reply:
PostgreSQL logs looks like this:

```
...
db-postgres-1  | 2022-03-03 03:15:54.405 UTC [69] LOG:  statement: 
db-postgres-1  | 	CREATE TABLE model_version_tags (
db-postgres-1  | 		key VARCHAR(250) NOT NULL, 
db-postgres-1  | 		value VARCHAR(5000), 
db-postgres-1  | 		name VARCHAR(256) NOT NULL, 
db-postgres-1  | 		version INTEGER NOT NULL, 
db-postgres-1  | 		CONSTRAINT model_version_tag_pk PRIMARY KEY (key, name, version), 
db-postgres-1  | 		FOREIGN KEY(name, version) REFERENCES model_versions (name, version) ON UPDATE cascade
db-postgres-1  | 	)
db-postgres-1  | 	
db-postgres-1  | 	
db-postgres-1  | 2022-03-03 03:15:54.410 UTC [69] LOG:  statement: ALTER TABLE model_versions ADD COLUMN run_link VARCHAR(500)
db-postgres-1  | 2022-03-03 03:15:54.412 UTC [69] LOG:  statement: ALTER TABLE model_versions ALTER COLUMN run_id DROP NOT NULL
db-postgres-1  | 2022-03-03 03:15:54.415 UTC [69] LOG:  statement: ALTER TABLE latest_metrics ALTER COLUMN is_nan TYPE BOOLEAN 
db-postgres-1  | 2022-03-03 03:15:54.415 UTC [69] LOG:  statement: ALTER TABLE latest_metrics ALTER COLUMN is_nan SET NOT NULL
db-postgres-1  | 2022-03-03 03:15:54.416 UTC [69] LOG:  statement: ALTER TABLE metrics ALTER COLUMN is_nan TYPE BOOLEAN 
db-postgres-1  | 2022-03-03 03:15:54.417 UTC [69] LOG:  statement: ALTER TABLE metrics ALTER COLUMN is_nan SET NOT NULL
👇👇👇👇👇👇👇👇👇👇👇👇👇👇👇👇
db-postgres-1  | 2022-03-03 03:15:54.420 UTC [69] LOG:  statement: CREATE INDEX index_params_run_uuid ON params (run_uuid)
db-postgres-1  | 2022-03-03 03:15:54.423 UTC [69] LOG:  statement: CREATE INDEX index_metrics_run_uuid ON metrics (run_uuid)
db-postgres-1  | 2022-03-03 03:15:54.425 UTC [69] LOG:  statement: CREATE INDEX index_latest_metrics_run_uuid ON latest_metrics (run_uuid)
db-postgres-1  | 2022-03-03 03:15:54.428 UTC [69] LOG:  statement: CREATE INDEX index_tags_run_uuid ON tags (run_uuid)
```

Reply:
Do we want to add an index on other columns?

Reply:
It certainly seems reasonable to create an index on experiment_id since we do have a number of joins employing that and a restrictive filter query on updates to the name. @dbczumar thoughts?